### PR TITLE
Improve profiler logging in the tracer

### PR DESCRIPTION
**Pull request by kevingosse**
## Summary of changes

- Don't try p/invoking into the profiler if it's not loaded (thus removing the warning in the logs)
- Lower the runtime id log to info
- Add values to the diagnostic log and the telemetry

## Reason for change

The extra logs were triggering Andrew's OCDs.

### CODE REVIEW

File: tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetryCollector.cs [hunk 1]
```diff
@@ -132,6 +133,8 @@ public ICollection<TelemetryValue> GetConfigurationData()
                 new(ConfigTelemetryData.FullTrustAppDomain, value: AppDomain.CurrentDomain.IsFullyTrusted),
                 new(ConfigTelemetryData.TraceMethods, value: settings.TraceMethods),
                 new(ConfigTelemetryData.ActivityListenerEnabled, value: settings.IsActivityListenerEnabled),
+                new(ConfigTelemetryData.ProfilerLoaded, value: Profiler.Instance.Status.IsProfilerReady),
+                new(ConfigTelemetryData.CodeHotspotsEnabled, value: Profiler.Instance.ContextTracker.IsEnabled),
```
Comment (andrewlock) [inline tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetryCollector.cs:137]: Do we have to use the static `Profiler.Instance`, this class is currently pretty testable 😿  I guess it doesn't really matter, but still

**Review by andrewlock — APPROVED** [review]

File: tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetryCollector.cs (hunk 1 above)
Comment (kevingosse) [reply tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetryCollector.cs]: Fixed in https://github.com/DataDog/dd-trace-dotnet/pull/2733/commits/d99bd1a11751f6a1b3a3d39712c910bb76a97a4e

**Review by andrewlock — APPROVED** [review]
Thanks for satisfying my OCD

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:warning: Merging #2733 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/e3be3b4d9da6d02902e0aa14025bea0e92fca211) will will **decrease** line coverage by `1%`
:warning: Merging #2733 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/e3be3b4d9da6d02902e0aa14025bea0e92fca211) will will **decrease** branch coverage by `1%`
:no_entry: Merging #2733 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/e3be3b4d9da6d02902e0aa14025bea0e92fca211) will will **increase** complexity by `253`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/e3be3b4d9da6d02902e0aa14025bea0e92fca211) | #2733       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `13651` / `19217` | `13667` / `19514` |          |
| Lines %   | `71%`      | `70%`      |  `-1%` :warning:  |
| Branches  | `7760` / `11403` | `7777` / `11587` |          |
| Branches %| `68%`      | `67%`      |  `-1%` :warning:  |
| Complexity|   `12690`      | `12943`        |  `253`  :no_entry:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=97370&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/97370/artifacts?artifactName=Code%20Coverage%20Report_97370&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2733](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=97511&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/97511/artifacts?artifactName=Code%20Coverage%20Report_97511&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :warning:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/e3be3b4d9da6d02902e0aa14025bea0e92fca211) | #2733       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `71%`      | `70%`       |  `-1%` :warning:  |
| Branches %| `68%`      | `67%`       |  `-1%` :warning:  |
| Complexity| `12690`      | `12943`       |  `253` :no_entry:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ContinuousProfiler.Profiler](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/ContinuousProfiler/Profiler.cs) | `-50%` :no_entry: | `0%` :heavy_check_mark:   | `2` :no_entry: |
| [Datadog.Trace.Telemetry.TelemetryController](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/Telemetry/TelemetryController.cs) | `-32%` :no_entry: | `-38%` :no_entry:   | `53` :no_entry: |
| [Datadog.Trace.TracerManager](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/TracerManager.cs) | `-28%` :no_entry: | `-31%` :no_entry:   | `88` :no_entry: |
| [Datadog.Trace.Telemetry.NullTelemetryController](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/Telemetry/NullTelemetryController.cs) | `-28%` :no_entry: | `0%` :heavy_check_mark:   | `10` :no_entry: |
| [Datadog.Trace.TracerManagerFactory](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/TracerManagerFactory.cs) | `-25%` :no_entry: | `-37%` :no_entry:   | `55` :no_entry: |
| [Datadog.Trace.Telemetry.ConfigurationTelemetryCollector](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/Telemetry/Collectors/ConfigurationTelemetryCollector.cs) | `-22%` :no_entry: | `-41%` :no_entry:   | `25` :no_entry: |
| [Datadog.Trace.Ci.GitInfo](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/Ci/GitInfo.cs) | `-17%` :no_entry: | `-11%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.ContinuousProfiler.ContextTracker](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/ContinuousProfiler/ContextTracker.cs) | `-8%` :no_entry: | `-15%` :no_entry:   | `13` :no_entry: |
| [Datadog.Trace.Ci.CIVisibility](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/Ci/CIVisibility.cs) | `7%` :heavy_check_mark: | `7%` :heavy_check_mark:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Ci.Processors.OriginTagTraceProcessor](https://github.com/DataDog/dd-trace-dotnet/tree/76cce2b550234d18449581d72decf3248327103d/tracer/src/Datadog.Trace/Ci/Processors/OriginTagTraceProcessor.cs) | `9%` :heavy_check_mark: | `17%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=97370&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/97370/artifacts?artifactName=Code%20Coverage%20Report_97370&api-version=6.0&%24format=zip)  
* [HTML report for this PR #2733](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=97511&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/97511/artifacts?artifactName=Code%20Coverage%20Report_97511&api-version=6.0&%24format=zip)

**Review by pierotibou — APPROVED** [review]

**Review by gleocadie — APPROVED** [review]
LGTM

**Pull request merged**